### ISSUE TITLE
Uncaught TypeError: Cannot read property 'appendChild' of null (google-maps.js: 12)

### ISSUE THREAD

**Issue by fvpDev** [dburles/meteor-google-maps#86]
Of all possible problems I am getting a `"Uncaught TypeError: Cannot read property 'appendChild' of null"` error for google-maps.js line 12 which is `document.body.appendChild(script);`

It's boggling my mind: I set it up to load within `Router.onAfterAction()` on the client...what could I be doing wrong, shouldn't `document.body` not return `null` at that point?

**Comment by fvpDev** [comment]
@jshimko, You ever encounter this problem? ^

**Comment by dburles** [comment]
Not sure exactly, I suggest trying `onBeforeAction` instead

**Comment by fvpDev** [comment]
Ok I will try that today/tomorrow

**Comment by dburles** [comment]
Hey @fvpDev interested to know if that solved the issue.

**Comment by fvpDev** [comment]
I called `GoogleMaps.ready()` in the `onCreated` hook of the map template without involving Router at all and I called `GoogleMaps.load()` in a `Meteor.startup()` call. I don't think I ever did manage to figure out what exactly the problem was but it turns out I didn't need to use Router.
Works for me -> http://joesnewyorkpizzeria.com/

That way, I was able to have everything gMaps-related in one file.
